src/qibo/validation/gate_set_tomography.py

```diff
@@ -0,0 +1,1273 @@
+
+    if gjk_1qb is None:
+        gjk_1qb = GST_1qb(nshots, noise_model, save_data=None)
+
+    nqubits = 1
+    class_qubit_gate = None
+    theta = None
+
+    BasisOps_13 = [
+        gates.I(0).matrix(backend=GlobalBackend()),
+        gates.X(0).matrix(backend=GlobalBackend()),
+        gates.Y(0).matrix(backend=GlobalBackend()),
+        gates.Z(0).matrix(backend=GlobalBackend()),
+        1 / np.sqrt(2) * np.array([[1, 1j], [1j, 1]]),
+        1 / np.sqrt(2) * np.array([[1, 1], [-1, 1]]),
+        1 / np.sqrt(2) * np.array([[(1 + 1j), 0], [0, (1 - 1j)]]),
+        1 / np.sqrt(2) * np.array([[1, -1j], [1j, -1]]),
+        gates.H(0).matrix(backend=GlobalBackend()),
+        1 / np.sqrt(2) * np.array([[0, (1 - 1j)], [(1 + 1j), 0]]),
+        np.array([[1, 0], [0, 0]]),
+        np.array([[0.5, 0.5], [0.5, 0.5]]),
+        np.array([[0.5, -0.5j], [0.5j, 0.5]]),
+    ]
+
+    ###################################
+    ### 13 scenarios # (2023-11-06) ###
+    ###################################
+
+    gatenames = (
+        "BasisOp00",
+        "BasisOp01",
+        "BasisOp02",
+        "BasisOp03",
+        "BasisOp04",
+        "BasisOp05",
+        "BasisOp06",
+        "BasisOp07",
+        "BasisOp08",
+        "BasisOp09",
+        "BasisOp10",
+        "BasisOp11",
+        "BasisOp12",
+    )
+    import time
+
+    tic = time.time()
+    idx_basis_ops = 0
+
+    Bjk_tilde_1qb = np.zeros((13, 4, 4))
+
+    for idx_basis_ops in range(0, 13):
+        print("GST 1qb 13 Basis Op. Basis Op idx = %d" % (idx_basis_ops))
+
+        # ===============#
+        # 13 SCENARIOS  # -- BIG ENDIAN NOTATION FOR THE 13 BASIS GATES TENSORED PRODUCTS
+        # ===============#
+
+        # Initialise 4 different initial states
+        for k in range(0, 4):
+            circ = prepare_1qb_state_new(k)
+
+            # ==================================================================================
+            # Basis operation 0 to 9
+            if idx_basis_ops < 10:
+                circ.add(
+                    gates.Unitary(
+                        BasisOps_13[idx_basis_ops],
+                        0,
+                        trainable=False,
+                        name="%s" % (gatenames[idx_basis_ops]),
+                    )
+                )
+
+            # ==================================================================================
+            # Basis operation 10, 11, 12
+            elif idx_basis_ops == 10:  # Reset, prepare |+> state
+                circ.add(
+                    gates.ResetChannel(0, [1, 0])
+                )  # ResetChannel(qubit, [prob0, prob1])
+                circ.add(gates.H(0))
+                circ.add(
+                    gates.Unitary(
+                        BasisOps_13[idx_basis_ops],
+                        0,
+                        trainable=False,
+                        name="%s" % (gatenames[idx_basis_ops]),
+                    )
+                )
```

Comment:
Also here, you are repeating the same thing, only slightly different, several times. You should change this the same way I asked for the functions above.

Reply:
Are you referring to the application of the reset channel?

Reply:
I believe that the unitaries implemented are different because they are chosen from ```BasisOps_13``` with index ```idx_basis_ops```. But I will take a look to verify that and merge them if they the same.

Reply:
The unitaries i implement are different. Also, to keep the code easy to decipher for the basis operations, I will leave these as
``` python
# Basis operation 10, 11, 12        
elif idx_basis_ops == 10: # Reset, prepare |+> state
    circ.add(gates.ResetChannel(0, [1,0])) # ResetChannel(qubit, [prob0, prob1])
    circ.add(gates.H(0))
    circ.add(gates.Unitary(BasisOps_13[idx_basis_ops], 0, trainable=False, name='%s' %(gatenames[idx_basis_ops])))

elif idx_basis_ops == 11:  # Reset, prepare |y+> state    
    circ.add(gates.ResetChannel(0, [1,0])) # ResetChannel(qubit, [prob0, prob1])
    circ.add(gates.H(0))
    circ.add(gates.S(0))
    circ.add(gates.Unitary(BasisOps_13[idx_basis_ops], 0, trainable=False, name='%s' %(gatenames[idx_basis_ops]))) 
    
elif idx_basis_ops == 12: # Reset, prepare |0> state
    circ.add(gates.ResetChannel(0, [1,0])) # ResetChannel(qubit, [prob0, prob1])
    circ.add(gates.Unitary(BasisOps_13[idx_basis_ops], 0, trainable=False, name='%s' %(gatenames[idx_basis_ops])))
```